File: src/main/java/com/ems/employee_management_system/service/AuthService.java
package com.ems.employee_management_system.service;

import com.ems.employee_management_system.dto.LoginRequest;
import com.ems.employee_management_system.dto.JwtResponse;
import com.ems.employee_management_system.dto.RegisterRequest;

/**
 * Service interface for authentication and registration operations.
 */
public interface AuthService {
    
    /**
     * Authenticates credentials and returns a signed JWT token with user metadata.
     */
    JwtResponse login(LoginRequest loginRequest);
    
    /**
     * Registers a new user account with encrypted passwords and roles assignment.
     */
    void register(RegisterRequest registerRequest);
}

File: frontend/src/services/authService.ts
import api from './api';
import type { LoginRequest, JwtResponse, RegisterRequest } from '../types';

export const authService = {
  login: (data: LoginRequest) =>
    api.post<JwtResponse>('/auth/login', data).then((r) => r.data),

  register: (data: RegisterRequest) =>
    api.post<{ message: string }>('/auth/register', data).then((r) => r.data),
};
